tik/lib/include/tik/CartographerKernel.h

```diff
@@ -10,17 +11,19 @@ namespace TraceAtlas::tik
     {
     public:
         CartographerKernel(std::vector<int64_t> basicBlocks, llvm::Module *M, std::string name = "");
+        std::vector<llvm::Value *> KernelImports;
+        std::vector<llvm::Value *> KernelExports;
 
     private:
         CartographerKernel();
-        void GetBoundaryValues(std::set<llvm::BasicBlock *> &blocks);
-        void BuildKernelFromBlocks(llvm::ValueToValueMapTy &VMap, std::set<llvm::BasicBlock *> &blocks);
+        void GetBoundaryValues(std::set<llvm::BasicBlock *> &blocks, std::map<int64_t, llvm::BasicBlock *> &IDToBlock);
+        void BuildKernelFromBlocks(llvm::ValueToValueMapTy &VMap, std::set<llvm::BasicBlock *> &blocks, std::map<int64_t, llvm::BasicBlock *> &IDToBlock);
         void InlineFunctionsFromBlocks(std::set<int64_t> &blocks);
```

Comment:
This dictionary that you are passing around everywhere, just make it global.

Reply:
Stand by this. Isn't this info global per module anyway?

Reply:
Changed. Declared at the top of CartographerKernel.cpp